crates/router/tests/connectors/utils.rs

```diff
@@ -69,6 +70,29 @@ pub trait ConnectorActions: Connector {
         call_connector(request, integration).await
     }
 
+    /// will retry the psync till the given status matches or retry max 3 times in a 10secs interval
+    async fn psync_retry_till_status_matches(
+        &self,
+        status: enums::AttemptStatus,
+        payment_data: Option<types::PaymentsSyncData>,
+        payment_info: Option<PaymentInfo>,
+    ) -> Result<types::PaymentsSyncRouterData, Report<ConnectorError>> {
+        let max_try = 3;
+        let mut curr_try = 1;
+        while curr_try <= max_try {
+            let sync_res = self
+                .sync_payment(payment_data.clone(), payment_info.clone())
+                .await
+                .unwrap();
+            if (sync_res.status == status) || (curr_try == max_try) {
+                return Ok(sync_res);
+            }
+            sleep(Duration::from_secs(10));
```

Comment:
10 secs is very big interval and we can't 30s to run a test. Reduce this to lesser frequency and use #[ignore] in the test.